app/forms/rules_of_use_form.rb

```diff
@@ -0,0 +1,43 @@
+class RulesOfUseForm
+  include ActiveModel::Model
+  include ActionView::Helpers::TranslationHelper
+
+  validate :validate_terms_accepted
+
+  attr_reader :terms_accepted
+
+  def self.model_name
+    ActiveModel::Name.new(self, nil, 'User')
+  end
+
+  def initialize(user)
+    @user = user
+  end
+
+  def validate_terms_accepted
+    return if @terms_accepted
+
+    errors.add(:terms_accepted, t('errors.rules_of_use'))
+  end
+
+  def submit(params, instructions = nil)
+    @terms_accepted = params[:terms_accepted] == 'true'
+    if valid?
+      process_successful_submission
+    else
+      self.success = false
+    end
+
+    FormResponse.new(success: success, errors: errors.messages)
+  end
+
+  private
+
+  attr_accessor :success, :user
+
+  def process_successful_submission
+    self.success = true
+    user.accepted_terms_at = Time.zone.now
+    user.save!
```

Comment:
at one point, we were trying to send all updates to user models through the `UpdateUser` service. WDYT of doing that here?